source/concepts/protocols.rst

```diff
@@ -0,0 +1,284 @@
+will be settled will be referred to ``parent layer`` in the further
+discussions.
+
+Ledger channel
+--------------
+
+   - Formed between any two parties having sufficient funds in their on-chain
+     accounts.
+   - The funds are directly deposited into the smart contracts on the blockchain.
+   - Allows any number of off-chain transactions between the participants.
+   - The channel must settled on the blockchain.
+
+Sub channel
+-----------
+
+   - Formed between participants who have a ledger channel established between
+     them.
+   - Funds are locked on the ledger channel. Hence no on-chain interaction
+     required to setup a channel.
+   - Channel is settled,
+
+     - By unlocking the funds from the ledger channel, if both parties agree on
+       the final state of the channel through off-chain transactions.
+     - By settling the parent channel on the blockchain, otherwise.
+
+       In this case, the parent channel and all other sub-channels opened using
+       the parent channel will also be settled.
+
+Virtual channel
+---------------
+
+   - Formed between two participants who do not have a ledger channel
+     established between them, but separately each of them have a ledger
+     channel established with an intermediary.
+   - Funds are locked in the two ledger channels that the participants have
+     with the intermediary. Hence, no on-chain interaction required.
+   - Channel is settled,
+
+     - By unlocking the funds from the two ledger channel, if both parties
+       agree on the final state of the channel through off-chain transactions.
+     - By settling the either one or two (depending on the scenario)
+       parent channel on the blockchain, otherwise.
+
+       In this case, the parent channels and all other sub-channels, virtual
+       channels opened using the parent channel will also be settled.
+
+.. note:
+
+   From the above descriptions, it can be seen that sub-channels and virtual
+   channels require **zero on-chain** interactions under normal circumstances.
+   On-chain interactions are required only when they do not agree on which
+   state should be settled.
+
+In the next section, protocols for corresponding to each of the four phases in
+the life cycle of a state channel are described.
+
+
+Phases of a state channel
+=========================
+
+Open phase
+----------
+
+In this phase, the channel is set up for off-chain transactions. Channel
+opening consists of two phases:
+
+1. Channel Proposal
+2. Funding
+
+The channel proposal protocol is same for all for three types of channels.
+However, the ``parent layer`` used for funding and the funding protocol are
+different.
+
+The ``parent layer`` in case of
+
+1. Ledger channel is blockchain.
+2. Sub-channel is ledger channel between the same participants.
+3. Virtual channel is two ledger channels: between each of the channel
+   participants and a common intermediary.
```

Comment:
```suggestion
different:

1. for ledger channels it is the blockchain,
2. for sub-channels it is the ledger channel between the same participants, and
3. for virtual channels the ``parent layer`` consists of the two ledger channels between each of the channel
   participants and a common intermediary.
```
```

Reply:
Regarding virtual channels: what about recursion, i.e. is it possible to construct a virtual channel on top of two virtual channels with a common intermediary, which in turn are based on a pair of ledger channels each?

Reply:
It is possible in principle (as per the description in the papers). But, with the current implementation it is not. Because, a virtual channel cannot hold locked funds.

@matthiasgeihs Your thoughts on this ?